streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -111,21 +229,27 @@
 
     private Map<TopicPartition, List<ConsumerRecord<byte[], byte[]>>> standbyRecords;
     private boolean processStandbyRecords = false;
-    private AtomicBoolean initialized = new AtomicBoolean(false);
 
     private ThreadCache cache;
 
     final ConsumerRebalanceListener rebalanceListener = new ConsumerRebalanceListener() {
         @Override
         public void onPartitionsAssigned(Collection<TopicPartition> assignment) {
             try {
-                log.info("stream-thread [{}] New partitions [{}] assigned at the end of consumer rebalance.",
+                if (state() == State.PENDING_SHUTDOWN) {
```

Comment:
As per comment above in `KafkaStreams` - you can access the `state` field directly